lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -280,11 +307,12 @@ def setup_access_token
         response['access_token']
       end
 
-      def commit(action, post, authorization = nil)
+      def commit(action, post, authorization = nil, method = :post)
         begin
-          raw_response = (action == :verify_payment ? ssl_get("#{base_url}/payments/#{post}", headers) : ssl_post(url(post, action, authorization), post.to_json, headers))
+          raw_response = (action == :verify_payment ? ssl_get("#{base_url}/payments/#{post}", headers) : ssl_request(method, url(action, authorization), post.to_json, headers))
```

Comment:
Sorry for coming back to this. I think the part of this statement, `"#{base_url}/payments/#{post}", headers`, can be added to the `def url` method as an additional switch/case?

Also, since we are providing a `method` can we just use `ssl_request`?